doc/README.md

```diff
@@ -0,0 +1,37 @@
+# Documentation
+
+## Objective
+
+The objective of this file is to describe how the build process of the documentation.
+
+## Tools
+
+We use `sphinx` to build our documentation, along with a few extensions.
+
+- `sphinx.ext.autodoc`
+
+  TODO: Explain to which files this applies to.
```

Comment:
Not the scope of this PR, I mainly wanted a place to write down some troubleshooting steps I found.